src/EnergyPlus/HybridModel.cc

```diff
@@ -142,29 +166,221 @@ namespace HybridModel {
+                    if (!FlagHybridModel_TM && FlagHybridModel_AI && FlagHybridModel_PC) {
+
+                        // !!! Consider: which variable to use?
 
-                    // Zone Air Infiltration Rate and Zone Internal Thermal Mass calculations cannot be performed simultaneously
-                    if (HybridModelZone(ZonePtr).InternalThermalMassCalc && HybridModelZone(ZonePtr).InfiltrationCalc) {
-                        HybridModelZone(ZonePtr).InfiltrationCalc = false;
+                        // Temperature and humidity ratio are measured
+                        if (TemperatureSchPtr > 0 && HumidityRatioSchPtr > 0){
+                            HybridModelZone(ZonePtr).InfiltrationCalc_T = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_T = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_H = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_H = true;
+                        }
+                        // Temperature and CO2 concentration are measured
+                        if (TemperatureSchPtr > 0 && CO2ConcentrationSchPtr > 0) {
+                            HybridModelZone(ZonePtr).InfiltrationCalc_T = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_T = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_C = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_C = true;
+                        }
+                        //  Humidity ratio and CO2 concentration are measured
+                        if (HumidityRatioSchPtr > 0 && CO2ConcentrationSchPtr > 0) {
+                            HybridModelZone(ZonePtr).InfiltrationCalc_H = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_H = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_C = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_C = true;
+                        }
+                    }
+
+					// Scenario 3: Three unknown parameters to solve
+					if (FlagHybridModel_TM && FlagHybridModel_AI && FlagHybridModel_PC) {
+						if (TemperatureSchPtr > 0 && HumidityRatioSchPtr > 0 && CO2ConcentrationSchPtr > 0) {
+                            HybridModelZone(ZonePtr).InternalThermalMassCalc_T = false;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_T = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_H = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_C = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_T = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_H = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_C = true;
+						}
+					}
+
+                    // Summarise hybridmodel flags
+                    if (HybridModelZone(ZonePtr).InternalThermalMassCalc_T){
```

Comment:
I don’t think the helper is required. If user defined arguments indicates YES, the input should be able to handle the case, otherwise should raise ERROR, using the above logic.

Reply:
@tsbyq